### PR TITLE
fix issue #20515 add-iso8601-basic-format-support

### PR DESCRIPTION
**Pull request by tanya654**
## Description
This Pull Request addresses the issue of limited support for the basic ISO 8601 date-time format in Trino, specifically extending the support to cover the basic format (e.g., "20010822T030405.321-1100"). This change ensures that the basic format is now supported in all scenarios where the extended format (e.g., "2001-08-22T03:04:05.321-11:00") was used.

This PR also references the closed issue [#13993](https://github.com/trinodb/trino/issues/13993), which added initial tests for extended ISO 8601 format but highlighted the need for additional coverage for the basic format.

This is part of addressing issue [#20515](https://github.com/trinodb/trino/issues/20515).

## Additional context and related issues
Added support for the basic ISO 8601 date-time format in functions like from_iso8601_timestamp. Now, both extended and basic formats are fully supported in scenarios where ISO 8601 date-time representation is used. ([#20515](https://github.com/trinodb/trino/issues/20515)).

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by tanya654** [comment]
> Thank you for your pull request and welcome to the Trino community. We require contributors to sign our [Contributor License Agreement](https://github.com/trinodb/cla/raw/master/Trino%20Foundation%20Individual%20CLA.pdf), and we don't seem to have you on file. Continue to work with us on the review and improvements in this PR, and submit the signed CLA to [cla@trino.io](mailto:cla@trino.io). Photos, scans, or digitally-signed PDF files are all suitable. Processing may take a few days. The CLA needs to be on file before we merge your changes. For more information, see https://github.com/trinodb/cla

Hi, I have sent the email to you.

**Comment by tanya654** [comment]
> Thank you for your pull request and welcome to the Trino community. We require contributors to sign our [Contributor License Agreement](https://github.com/trinodb/cla/raw/master/Trino%20Foundation%20Individual%20CLA.pdf), and we don't seem to have you on file. Continue to work with us on the review and improvements in this PR, and submit the signed CLA to [cla@trino.io](mailto:cla@trino.io). Photos, scans, or digitally-signed PDF files are all suitable. Processing may take a few days. The CLA needs to be on file before we merge your changes. For more information, see https://github.com/trinodb/cla

Hi, i have sent the email to you several days ago.

File: core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java [hunk 1]
```diff
@@ -242,22 +242,48 @@ public static Slice toISO8601FromDate(@SqlType(StandardTypes.DATE) long date)
...
 
-    @ScalarFunction("from_iso8601_timestamp")
-    @LiteralParameters("x")
-    @SqlType("timestamp(3) with time zone")
-    public static long fromISO8601Timestamp(ConnectorSession session, @SqlType("varchar(x)") Slice iso8601DateTime)
-    {
-        DateTimeFormatter formatter = ISODateTimeFormat.dateTimeParser()
-                .withChronology(getChronology(session.getTimeZoneKey()))
-                .withOffsetParsed();
+@ScalarFunction("from_iso8601_timestamp")
```
Comment (ebyhr) [inline core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java:245]: Please fix indentation. https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#code-style

File: core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java [hunk 2]
```diff
@@ -242,22 +242,48 @@ public static Slice toISO8601FromDate(@SqlType(StandardTypes.DATE) long date)
...
+            return packDateTimeWithZone(parseDateTimeHelper(basicFormatter, iso8601DateTime.toStringUtf8()));
         }
-        catch (IllegalArgumentException e) {
+        catch (IllegalArgumentException innerException) {
             throw new TrinoException(INVALID_FUNCTION_ARGUMENT, e);
         }
     }
 
+}
+
```
Comment (ebyhr) [inline core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java:286]: Please add tests to `TestDateTimeFunctions`.

File: core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java (hunk 1 above)
Comment (ebyhr) [inline core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java:245]: > fix issue trinodb#20515 add-iso8601-basic-format-support

https://trino.io/development/process#pull-request-and-commit-guidelines- is a commit message guideline in this project. Please rename to `Add support for basic format in from_iso8601_timestamp function` or something.

**Comment by ebyhr** [comment]
@cla-bot check

**Comment by mosabua** [comment]
Please reopen if you want to continue working on this PR @tanya654

**Pull request closed without merging**